### PR TITLE
Enable upgrade to PHP 8.0

### PR DESCRIPTION
**Pull request by webdevel**

### CODE REVIEW

**Comment by adrienbrault** [comment]
The CI needs to be updated to also test php 8 before this can be merged (there might be failing tests to address).

I would do that in `.travis.yml`, but I'm surprised to see that travis did not run on this PR? Maybe there's more work to be done to get the CI back up and running.

**Comment by webdevel** [comment]
> The CI needs to be updated to also test php 8 before this can be merged (there might be failing tests to address).
> 
> I would do that in `.travis.yml`, but I'm surprised to see that travis did not run on this PR? Maybe there's more work to be done to get the CI back up and running.

Screen Shot 2021-02-09 at 8 21 17 PM

[build passing](https://travis-ci.com/github/webdevel/string-matcher)

File: composer.json [hunk 1]
```diff
@@ -16,23 +16,26 @@
             "type": "vcs",
             "url": "https://github.com/sourceability/coding-standard",
             "no-api": true
+        },
+        {
+            "type": "vcs",
+            "url": "git@github.com:webdevel/liquidmetal-php.git"
```
Comment (adrienbrault) [inline composer.json]: Why is this needed?

File: examples/builder_example.php [hunk 2]
```diff
@@ -9,6 +9,8 @@
 use Sourceability\StringMatcher\Simplifier\ChainSimplifier;
 use Sourceability\StringMatcher\Simplifier\TrimmedLowercaseSimplifier;
 use Sourceability\StringMatcher\ValuesMatcherBuilder;
+use const PHP_EOL;
+use function print_r;
```
Comment (adrienbrault) [inline examples/builder_example.php:13]: I think all the changes to `examples/` can be removed from the diff right?

File: phpstan.neon [hunk 3]
```diff
@@ -5,12 +5,7 @@ includes:
     - vendor/phpstan/phpstan-beberlei-assert/extension.neon
     - vendor/phpstan/phpstan-deprecation-rules/rules.neon
 
-
 rules:
-    - Sourceability\CodingStandard\PHPStan\Rules\ConstantBooleanIdenticalRule
```
Comment (adrienbrault) [inline phpstan.neon:10]: Did you run into issues with phpstan? I would like to understand why those rules were removed?

File: composer.json (hunk 1 above)
Comment (webdevel) [reply composer.json]: > Support for deprecated curly braces for offset access has been removed. 

- https://www.php.net/manual/en/migration80.incompatible.php

File: examples/builder_example.php (hunk 2 above)
Comment (webdevel) [reply examples/builder_example.php:13]: Screen Shot 2021-02-10 at 9 50 26 AM

No, we should not remove these use statements. These were added because the examples are using those in the global namespace. And the examples are CS checked.

File: phpstan.neon (hunk 3 above)
Comment (webdevel) [reply phpstan.neon:10]: Screen Shot 2021-02-10 at 9 57 43 AM

As far as I can tell, those rules do not exist.

Comment (mbadolato) [reply phpstan.neon:10]: Maybe something isn't importing correctly?

https://github.com/sourceability/phpstan-rules/tree/master/src/Rules
Screen Shot 2021-02-10 at 11 47 32 AM

Comment (webdevel) [reply phpstan.neon:10]: That dependency `sourceability/phpstan-rules` is currently not required by this project. I will add it then.

Comment (mbadolato) [reply phpstan.neon:10]: AH! I probably did the same thing 😆 I linked to the proper repo `phpstan-rules` but saw `"sourceability/coding-standard": "^1.0@dev",1 in `composer.json` and just glossed over it as "The standard are there" Sorry bout that!

**Comment by webdevel** [comment]
Screen Shot 2021-02-10 at 12 13 30 PM
 Latest build is passing.

File: composer.json (hunk 1 above)
Comment (webdevel) [reply composer.json]: Even though upstream has already merged my changes (but not tagged for release), removing my fork causes build to fail. For whatever reason, using upstream master branch does not have the latest changes. This could easily be removed when upstream tags and releases. I do not think we should make this a blocker.

Comment (adrienbrault) [reply composer.json]: This is a blocker to merging this PR IMHO.

Any project that will want to use this string-matcher patch, will need to also add `git@github.com:webdevel/liquidmetal-php.git` to the project's composer repositories like you did here. So, might as well just add both `git@github.com:webdevel/liquidmetal-php.git` and `git@github.com:webdevel/string-matcher.git` to your project for now.

Once the need to add `git@github.com:webdevel/liquidmetal-php.git` (along with the `dev-master` version constraint, which is also a blocker imho) is dropped, then this PR can be merged, and a tag created on master.

Comment (webdevel) [reply composer.json]: Point taken. I have removed VCS, tagged, published, released my fork and adopted long-term maintenance (https://packagist.org/packages/webdevel/liquidmetal-php) as previous author/maintainer went 6 years without update.
Screen Shot 2021-02-11 at 10 18 11 AM
[Latest build is passing](https://travis-ci.com/github/webdevel/string-matcher)

**Comment by adrienbrault** [comment]
Thank you!

**Comment by adrienbrault** [comment]
Tagged [0.2.0](https://github.com/sourceability/string-matcher/releases/tag/0.2.0)
